### PR TITLE
Flow Blueprints mvp tidy

### PR DESCRIPTION
**Pull request by knolleary**
## Description

This applies some final tidy up to the first iteration of Flow Blueprints work - #2828

I've split it into separate commits

 1. Include `instance-blueprint` in the payload sent to posthog when creating an instance
 2. Updates the Blank template wording:
    image
 3. Updates all external apis to use `flow blueprint` terminology rather than flow template.
     - CRUD end-points moved to `/api/v1/flow-blueprints` and scheme updated
     - Property on Instance Create api renamed to `flowBlueprintId`
     - It does *not* modify the model/database table - that felt a step too far for today.

### CODE REVIEW

File: frontend/src/api/instances.js [hunk 1]
```diff
@@ -9,15 +9,15 @@ import paginateUrl from '../utils/paginateUrl.js'
 import client from './client.js'
 
 const create = async (options) => {
-    // Backend calls these flow templates, UI refers to it as flow blueprints
-    options.flowTemplateId = options.flowBlueprintId
-    delete options.flowBlueprintId
-
+    if (options.flowBlueprintId === '') {
+        delete options.flowBlueprintId
```
Comment (Pezmc) [inline frontend/src/api/instances.js:13]: Does the backend not handle receiving ''?  I think it should!

File: frontend/src/pages/instance/components/InstanceForm.vue [hunk 2]
```diff
@@ -11,7 +11,7 @@
         <!-- Form title -->
         <div class="mb-8 text-sm text-gray-500">
             <template v-if="creatingNew">
-                <template v-if="applicationSelection">
+                <template v-if="!creatingApplication || applicationSelection">
```
Comment (Pezmc) [inline frontend/src/pages/instance/components/InstanceForm.vue:14]: Is this change intended?

File: frontend/src/api/instances.js (hunk 1 above)
Comment (knolleary) [reply frontend/src/api/instances.js:13]: It does handle it. I just dislike sending blank properties if they are meant to be ignored.

File: frontend/src/pages/instance/components/InstanceForm.vue (hunk 2 above)
Comment (knolleary) [reply frontend/src/pages/instance/components/InstanceForm.vue:14]: Yes - albeit slightly off-topic and it slipped in.

I noticed when we're on the 'Create Instance' page (so no Application creation/selection to be done), we were showing the 'Applications are used to ...' explainer. This fixes that to show the header related to creating an instance.
